src/chromium/interfaces/getters.js

```diff
@@ -1,19 +1,29 @@
 import { getExternalBrowser } from "Shared/backgroundScripts/getters.js";
 
+const nativeApps = [
+  "org.mozilla.firefox_bridge_nmh_dev",
+  "org.mozilla.firefox_bridge_nmh_nightly",
+  "org.mozilla.firefox_bridge_nmh",
+  "org.mozilla.firefox_bridge_nmh_esr",
+];
+
 /**
- * Determines whether Firefox is installed on the system.
+ * Determines whether Firefox is installed on the system and returns the name of the installed Firefox variant in
+ * order of preference: dev, nightly, release, esr.
  *
- * @returns {Promise<boolean>} True if Firefox is installed on the system, false otherwise.
+ * @returns {Promise<string>} The name of the installed Firefox variant, or undefined if a Firefox variant is not installed.
  */
-export async function getIsFirefoxInstalled() {
-  // NOTE: We are not currently handling the case where Firefox is not installed until
-  // native messaging is implemented. We will assume for now that Firefox is always installed.
-  const result = await browser.storage.session.get("isFirefoxInstalled");
-  if (result.isFirefoxInstalled === undefined) {
-    browser.storage.session.set({ isFirefoxInstalled: true });
-    return true;
+export async function getInstalledFirefoxVariant() {
+  // Check for installed variants using the respective native messaging host.
+  for (const nativeApp of nativeApps) {
```

Comment:
Is the latency acceptable?

If not, consider caching the result in the storage.session API.

A potential issue with caching the availability is that the availability of the browser may change at runtime. A way to counter that is to try the most recently found browser first.

Reply:
It appears like it is. How would I figure this out? Populating this list with 100 nmhs still remains instantaneous.

Reply:
What Rob suggests is probably for the best:

```
let previousNativeApp = await browser.settings.local.get("WHATEVER");

const isValid = async (nativeApp) => {
  try {
    await browser.runtime.sendNativeMessage(nativeApp, {command: "GetVersion",});
    return true;
  } catch (err) {
    return false;
  }
};

if (await isValid(previousNativeApp)) {
  return previousNativeApp;
}

// Check for installed variants using the respective native messaging host.
for (const nativeApp of nativeApps) {
  if (await isValid(nativeApp)) {
    await browser.settings.local.set("...", nativeApp);
    return nativeApp;
  }
}

return undefined;
```